### PR TITLE
`azurerm_mysql_flexible_server` - support `geo_backup_key_vault_key_id` and `geo_backup_user_assigned_identity_id`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/20408

To enable `geo_redundant_backup_enabled`, these two properties `geo_backup_key_vault_key_id` and `geo_backup_user_assigned_identity_id` also need to be set.

image

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Test failures:
```
------- Stdout: -------
=== RUN   TestAccMySqlFlexibleServer_complete
=== PAUSE TestAccMySqlFlexibleServer_complete
=== CONT  TestAccMySqlFlexibleServer_complete
    testcase.go:110: Step 1/2 error: Error running apply: exit status 1
        
        Error: creating Flexible Server (Subscription: "*******"
        Resource Group Name: "acctestRG-mysql-230307035443686329"
        Flexible Server Name: "acctest-fs-230307035443686329"): polling after Create: Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: '60d8c15a-3b56-48bd-9726-18b7f576fa67'"
        
          with azurerm_mysql_flexible_server.test,
          on terraform_plugin_test.tf line 63, in resource "azurerm_mysql_flexible_server" "test":
          63: resource "azurerm_mysql_flexible_server" "test" {
        
--- FAIL: TestAccMySqlFlexibleServer_complete (544.95s)
FAIL
```

**Comment by neil-yechenwei** [comment]
@katbyte , due to quota, I changed the test location for mysql in settings.kt in this PR. So maybe you also have to change the test location before PR merge. I just now reran the test cases. Seems they all passed.

image

**Comment by katbyte** [comment]
@neil-yechenwei - if the wrong location is picked `Code="InternalServerError" Message="An unexpected error occured while processing the request. ` is the error?

if that is so could you please open an issue on the rest specs detailing it, and adding a note in the documentation as that is a very hard to action/understand error

**Comment by neil-yechenwei** [comment]
@katbyte , raised an [issue](https://github.com/Azure/azure-rest-api-specs/issues/22980) on azure rest api spec. Actually, the test case you mentioned has failed for a long time. It's not related with this PR.

**Comment by neil-yechenwei** [comment]
@katbyte , Seems the quota issue is resolved on your test subs. So I reverted the change about test location. I just now reran test cases and all passed.

image
image

**Comment by neil-yechenwei** [comment]
Hi @katbyte , as test cases passed, is it fine to be merged for this PR?

File: internal/services/mysql/mysql_flexible_server_resource_test.go [hunk 1]
```diff
@@ -463,6 +463,21 @@ func TestAccMySqlFlexibleServer_updateToCustomerManagedKey(t *testing.T) {
 	})
 }
 
+func TestAccMySqlFlexibleServer_enableGeoRedundantBackup(t *testing.T) {
```
Comment (katbyte) [inline internal/services/mysql/mysql_flexible_server_resource_test.go:466]:
```suggestion

// this test can fail with a uninformative error, tracked here https://github.com/Azure/azure-rest-api-specs/issues/22980
func TestAccMySqlFlexibleServer_enableGeoRedundantBackup(t *testing.T) {
```

**Review by katbyte — APPROVED** [review]
Thanks, LGTM aside from 1 comment

**Pull request merged**